### PR TITLE
Create a better defined ElasticTransport

### PR DESCRIPTION
**Pull request by barlock**
**Related Issue**  
Supports #1 

**Related PRs**  
This PR is not dependent on any other PR

**What does this PR do?**  
Adds an Elastic transport that will later be used to pick up logging from where it left off. 

**Description of Changes**  
Better defined Transport interface. 

**What gif most accurately describes how I feel towards this PR?**

### CODE REVIEW

File: packages/github-collector/docker-compose.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 elk:
-  image: sebp/elk
+  image: sebp/elk:643
```
Comment (barlock) [inline packages/github-collector/docker-compose.yml]: Locking down version to what's on AWS

File: packages/github-collector/src/ElasticTransport.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  }
+
+  /**
+   * Gets the last event of a specific type that was logged
+   *
+   * @param {string} eventType - The event type logged
+   * @param {Object} query - Any addtional key/value pairs to search for
+   * @returns {Promise<Object>} The event
+   */
+  async lastEventOfType(eventType, query = {}) {
```
Comment (barlock) [inline packages/github-collector/src/ElasticTransport.js]: Will be used later in something similar to #15

File: packages/github-collector/test/TestTransport.js [hunk 3]
```diff
@@ -21,4 +25,38 @@ module.exports = class TestTransport extends Transport {
...
+   * Create a new TestTransport
+   */
+  constructor() {
+    this._transport = new TestWinstonTransport();
+
+    this.logger = winston.createLogger({ transports: [this._transport] });
+  }
+
+  /**
+   * Get the logs that were logged by the logger
```
Comment (breakpointer) [inline packages/github-collector/test/TestTransport.js]: Such log!

**Comment by barlock** [comment]
LOGS!!!

**Pull request merged**